common/utils/retry.go

```diff
@@ -0,0 +1,40 @@
+package utils
+
+import (
+	"time"
+
+	"github.com/quintilesims/layer0/common/errors"
+)
+
+func Retry(timeout, delay time.Duration, fn func() (bool, error)) error {
```

Comment:
I like this, very straightforward. Slight nit but how do ppl feel about renaming `delay` to `delayInterval`? Otherwise I read 'delay' as the delay before `Retry()` initially executes, which doesn't imply looping (to me, at least).

Reply:
yeah `interval` works for me